presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/ReadOnlySystemAccessControl.java

```diff
@@ -143,4 +144,9 @@ public void checkCanShowSchemas(SystemSecurityContext context, String catalogNam
     public void checkCanShowRoles(SystemSecurityContext context, String catalogName)
     {
     }
+
+    @Override
+    public void checkCanExecuteProcedure(SystemSecurityContext systemSecurityContext, CatalogSchemaProcedureName procedure)
+    {
+    }
```

Comment:
it should deny; procedures are purely side-effect invocations, so arguably none is read-only operation